components/scheduler/src/Scheduler.svelte

```diff
@@ -453,8 +380,37 @@
           </li>
         {/each}
       </ul>
-      <button class="book" on:click={() => bookTimeSlots(slotsToBook)}
-        >{booking_label}</button
+      {#if custom_fields.length}
+        <div id="custom-fields">
+          {#each custom_fields as field}
+            {#if field.type === "email"}
+              <label data-required={field.required}>
+                <strong>{field.title}</strong>
+                <!-- TODO: see if we can make type="text" dynamic for email case. Svelte doesnt care for it as is. -->
+                <input
+                  type="email"
+                  bind:value={customFieldResponses[field.title]}
+                />
+              </label>
+            {:else}
+              <label data-required={field.required}>
+                <strong>{field.title}</strong>
+                <input
+                  type="text"
+                  bind:value={customFieldResponses[field.title]}
+                />
+              </label>
+            {/if}
+          {/each}
+        </div>
+      {/if}
+      <button
+        disabled={!requiredFieldsFilled}
+        title={!requiredFieldsFilled
```

Comment:
What is the use of `title` attribute in this html button element?

Reply:
@pooja169usp this makes it so, when hovering the disabled button, it will remind you that the rason it's diabled is that you have not filled out all fields.